src/deepsparse/image_classification/pipelines.py

```diff
@@ -76,12 +81,18 @@ def __init__(
         top_k: int = 1,
         **kwargs,
     ):
+        self.config = None
+
         super().__init__(**kwargs)
 
         if isinstance(class_names, str) and class_names.endswith(".json"):
             self._class_names = json.load(open(class_names))
         elif isinstance(class_names, dict):
             self._class_names = class_names
+        elif self.config and LABELS_TO_CLASS_MAPPING_NAME in self.config:
+            self._class_names = {
```

Comment:
let's get rid of `class_names` and just make this `config` (similar to how `transformers` treats their config) - too much room for casework/overwriting here

Reply:
class_names under the hood